crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -137,6 +137,8 @@ impl CommittedState {
         let mut tx_data = TxData { records: vec![] };
         for (table_id, table) in tx_state.insert_tables {
             let commit_table = self.get_or_create_table(table_id, &table.row_type, &table.schema);
+            // The schema may have been modified in the transaction.
+            commit_table.schema = table.schema;
```

Comment:
Can you add a test case to the bottom of this file which tests for this specific case?

Reply:
Can do -- but are you implying that this is the right thing to do? Also what about `row_type`, would this not be mutable too if we introduced DDL operations?

Reply:
Yes, this is the right thing to do right now based on how we have it set up, and yes it would also have to be mutable if we introduced DDL operations. But I think in general we'd have to redesign the LockingTxDatastore. My implementation was always meant to be temporary tbh until we got the time to design a proper one.